lib/pid_manager/src/pid_manager.cpp

```diff
@@ -20,7 +22,7 @@ struct string {
 
 void init_string(struct string *s) {
     s->len = 0;
-    s->ptr = malloc(s->len+1);
+    s->ptr = reinterpret_cast<char*>(malloc(s->len+1));
```

Comment:
Why not `static_cast<char*>`? Same question for the other new `reinterpret_cast<char*>` uses in this file.